### PR TITLE
remove aec_db_error_store

### PR DESCRIPTION
**Pull request by rustaem**
*_activity and ensure_transaction fixed. Suptree and it's diagram fixed. Tests fixed. aec_db_gc fixed just a bit.

Addressing #3682.

### CODE REVIEW

**Comment by cytadela8** [comment]
This should be rebased on master

File: apps/aehyperchains/test/aehc_parent_mng_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,249 @@
+%%% -*- erlang-indent-level: 4 -*-
```
Comment (cytadela8) [inline apps/aehyperchains/test/aehc_parent_mng_SUITE.erl:1]: This file shouldn't  be here ;)

File: apps/aehyperchains/test/aehc_test_utils.erl [hunk 2]
```diff
@@ -0,0 +1,119 @@
+-module(aehc_test_utils).
```
Comment (cytadela8) [inline apps/aehyperchains/test/aehc_test_utils.erl:1]: This file also shouldn't be here

**Review by cytadela8 — DISMISSED** [review]

**Comment by cytadela8** [comment]
@rustaem See why the tests are failing. The code looks overall fine now.

**Comment by rustaem** [comment]
> @rustaem See why the tests are failing. The code looks overall fine now.

Will do.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1268,8 +1268,8 @@ jobs:
           name: Test DB compatibility
           command: |
             cp -r ./node_stable/data/mnesia ./node_new/data/
-            ./node_new/bin/aeternity start
-            sleep 30
+            ./node_new/bin/aeternity daemon
```
Comment (cytadela8) [inline .circleci/config.yml:1271]: This is probably not needed in the end

File: apps/aecore/src/aec_db.erl [hunk 4]
```diff
@@ -591,78 +579,73 @@ find_discovered_pof(Hash) ->
...
-       [{aec_chain_state, top_block_node}]).
+    State = #aec_chain_state{
+        key = top_block_node,
+        value = #{
+            hash => Hash,
+            header => Hdr
+        }
+    },
+    ?t(mnesia:write(State)).
+%%?t(mnesia:write(#aec_chain_state{key = top_block_node, value = #{ hash => Hash
```
Comment (cytadela8) [inline apps/aecore/src/aec_db.erl:637]: Remove commented code

**Review by cytadela8 — APPROVED** [review]

File: apps/aecore/test/aec_chain_tests.erl [hunk 5]
```diff
@@ -714,11 +714,15 @@ fork_common_block(EasyChain, TopHashEasy, HardChain, TopHashHard) ->
...
 fork_test_chain_ends_and_migration(ExpectedTops, ExpectedBlocks) ->
-    F = fun() -> ?assertEqual( lists:usort(ExpectedTops)
-                             , lists:usort(aec_db:find_chain_end_hashes())),
-                 ?assertEqual( [ aec_blocks:to_header(Block) || Block <- ExpectedBlocks]
-                             , [ header_by_height(H) ||
-                                 H <- lists:seq(0, aec_blocks:height(lists:last(ExpectedBlocks)))])
+    ExpectedTopsSorted = lists:usort(ExpectedTops),
+    F =
+        fun() ->
+            EndHashesSorted = lists:usort(aec_db:find_chain_end_hashes()),
```
Comment (cytadela8) [inline apps/aecore/test/aec_chain_tests.erl:720]: Btw, This is less readable for me, but whatever.

**Review by cytadela8 — APPROVED** [review]
Thanks @rustaem 

@uwiger This is ready for when we update mnesia-rocksdb plugin.

**Comment by uwiger** [comment]
Closing. Similar work done in #3777 (and some changes copied from here).

**Pull request closed without merging**